File: programmers/python 3/42587.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/42587

def solution(priorities, location):
    priorities = [(i, p) for i, p in enumerate(priorities)]
    count = 0
    while priorities:
        J = priorities.pop(0)
        if any(p[1] > J[1] for p in priorities):
            priorities.append(J)
        else:
            count += 1
            if J[0] == location:
                break
    return count

File: programmers/swift/42587.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/42587

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var priorities = priorities.enumerated().map { $0 }
    var count = 0

    while !priorities.isEmpty {
        let J = priorities.removeFirst()
        if priorities.contains(where: { $0.element > J.element }) {
            priorities.append(J)
        } else {
            count += 1
            if J.offset == location {
                break
            }
        }
    }

    return count
}